examples/dex.py

```
@@ -0,0 +1,90 @@
+    storage['token_reserve'] = new_token_reserve
+    
+    # Credit tokens to buyer
+    sender = msg['sender']
+    storage[sender] = storage.get(sender, 0) + tokens_out
+
+elif msg['data'].startswith('sell:'):
+    # User sells DEX tokens to get native coins back
+    if storage.get('k') is None:
+        raise Exception("Not initialized")
+        
+    parts = msg['data'].split(':')
```

Ruff (0.15.15)

[error] Undefined name `msg`

(F821)